### ISSUE TITLE
Normalize atoms at insertion time

### ISSUE THREAD

**Issue by ngeiswei** [opencog/atomspace#1778]
As suggested by @vsbogd in https://github.com/opencog/atomspace/pull/1774#issuecomment-400259218 it could be good to have some level of normalization at insertion time (whenever it makes sense, i.e. the non-normalized form is never desired).

For instance, inserting in the atomspace
```
(Scope (Variable "$X") (Variable "$X"))
```
then add/get the equivalent scope
```
(Scope (Variable "$X"))
```
returns
```
(Scope (Variable "$X") (Variable "$X"))
```

Such normalization would automatically remove the unnecessary variable declaration (since the variable are unconstrained) at insertion time.

Note: there is some form of normalization at execution time in `FoldLink::delta_reduce` (and its heirs) but none at insertion time.

**Comment by linas** [comment]
Due to typing, some normalization is non-obvious.  For example, `(TypedVariable (Glob "$x") (Interval (Number 1)(Number 1)))` is the same as `(Variable "$X")`  .. or is supposed to be the same ... the code probably treats them differently, although at run-time, they should have the same results.

**Comment by ngeiswei** [comment]
Yes, I think automatic normalization may only concern basic equivalences, like the ones currently hard wired to detect scope link alpha-conversion. For more sophisticated stuff the URE should probably be used, although obviously the line is rather blurry.

**Comment by linas** [comment]
automatic normalization is discussed here: https://github.com/opencog/atomspace/issues/1763#issuecomment-399557144
